### DISCUSSION TITLE
0062 XLS-62d: Options

### DISCUSSION THREAD

**Discussion by dangell7** [XRPLF/XRPL-Standards#186] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by xVet** [comment]
Great proposal Denis

**Comment by desimmons** [comment]
It the proposal, you say "Five new transaction types" but then only list three.

**Reply by dangell7** [reply]
Updated thank you

**Comment by desimmons** [comment]
IIUC, this draft doesn't consider American style options. Is that right?

**Reply by dangell7** [reply]
This draft is for American style options where exercise can be done anytime before expiration. European-style options can only be exercised at the expiration date of the option.

**Reply by desimmons** [reply]
Was trying to find statistics (didn't spend much): I can see that American style options are much more popular (for obvious reasons!?) than European. But would we want to accommodate European options at some point? If so, would European style options be implemented as a separate amendment or capture that as part of this amendment?

**Reply by dangell7** [reply]
I can do it in the ListOption with a flag. I think we should include both types. 

Thanks for the input!

**Comment by shortthefomo** [comment]
From my going over this everything in the disclaimer is left to the interface implementing the protocol in their interface correct? none of those issues highlighted there could be addressed on ledger?

Maybe something could be looked at to leverage the DiD to "check" and flag some of those issue. eg, requiring a DiD that has the accreditation once that is added only then the option to use options becomes available.

**Reply by dangell7** [reply]
Correct. The DID/Credential is something that was introduced after this proposal and could be implemented.

**Comment by shortthefomo** [comment]
Another question reading over this once again is the **Expiration** is this yet another case where a standard timestamp is used and not ripple time stamp 🫣

Looking at other documentation it seems to be standard to offer fixed limits to this expiration? What happens in the case someone picks a date 1000 years out.

- 30 days
- 60 days
- 90 days
- One year 

(Ah understanding the American vs European style covers this)

**Reply by dangell7** [reply]
Expiration is typically the third Friday of each month. There can be options with any expiration but if you want TradeFi then those listing options, would likely follow the regular guidelines as those with weird option expiration would never get any open interest.

As for ripple/unix I don't care, I would prob do ripple time and let the client convert.

**Comment by fabiansinani203** [comment]
A great proposal Denis! I would like to see it to be tested and progressed aswell.

**Comment by shortthefomo** [comment]
Strike price also is singular here is offering multiple strike prices not possible or something that could be worked on later as a follow up?

**Reply by dangell7** [reply]
Multiple prices is possible. You can create the entire option table with this.

However, the strike price is quoted in one currency. I would use RLUSD. You could do multiple but without pathing or some added functionality the Option in JPN would never cross the Option in USD.

Also, as a guideline, the option should always be quoted in the currency that is used to purchase the asset. So if its an US stock option then USD.

**Reply by shortthefomo** [reply]
I missed understanding that nice.


These are not active but passive objects as I understand, right so there is nothing monitoring when the option can be exorcised but rather that is required from a user to 'prompt' the execution of it via OptionExercise.


What happens in the case both parties never exorcise it? Is there a way like with escrows anyone could prod it and it would execute?


Agree with its up to participant to convert it to other currency if that is what they so chose after its execution.

**Comment by dangell7** [comment]
Updated:

- [x] Option Order Book
- [x] Automated Option Offer Matching/Swapping (Partial Contracts)
- [x] RPC Commands
- [x] Delete List Transaction (Deletes the entire option after expiration)
